src/Common/src/System/Net/Security/CertificateValidation.Unix.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Globalization;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Security
+{
+    internal static class CertificateValidation
+    {
+        internal static SslPolicyErrors BuildChainAndVerifyProperties(X509Chain chain, X509Certificate2 remoteCertificate, bool checkCertName, string hostName)
+        {
+            SslPolicyErrors errors = chain.Build(remoteCertificate) ?
+                SslPolicyErrors.None :
+                SslPolicyErrors.RemoteCertificateChainErrors;
+
+            if (!checkCertName)
+            {
+                return errors;
+            }
+
+            if (string.IsNullOrEmpty(hostName))
+            {
+                return errors | SslPolicyErrors.RemoteCertificateNameMismatch;
+            }
+
+            int hostNameMatch;
+            using (SafeX509Handle certHandle = Interop.Crypto.X509Duplicate(remoteCertificate.Handle))
```

Comment:
Which would still need to then get the address bytes from the IP or convert the string to ascii, and then marshal those to native code.  Less cost, but still cost.